strava.py

```diff
@@ -66,16 +65,18 @@ def segment_leaderboard(self, segment_id, token, club_id, date_range):
                          for entry in leaderboard]
         return leader_result
 
-    # Gather hop segment leaders for a given date
-    def hop_alltime_leaders(self, db):
-
-        #TODO: Move this kind of stuff into DB service
-        john = db['326452']
-        johns_token = self.get_user_access_token(john, db)
+    def hop_alltime_leaders(self):
+        """Gather all-time hop segment leaders."""
+        # TODO: clarify if we really should use John for this.  Couldn't
+        # this use the current user's token?  Maybe that would be slightly
+        # safer for rate limits?
+        john = models.User.get('326452')
+        johns_token = self.get_user_access_token(john)
```

Comment:
In order to leverage the user's token we would need to authenticate them so we knew who they were. They'll only be registering once so we can't really rely on that to initiate a session/etc reliably. Were you thinking of some sort of approach in particular?

I think a good approach would be to use the token that was issued to my API Account `Segmund`. The refresh token is currently stored in config.json and we would just need to keep the auth token and expiration time somewhere. It looks like the rate limiting is tied to the Segmund app anyways regardless of what token we're using, likely via client_id. I can see the rate usage data in Strava.

Reply:
Hmm.  They're only registering once, but we check the validity of the access token on every request and get new tokens if necessary.  Assuming it's a user driving the browser, aren't we guaranteed to have a good set of tokens?

Things that happen without user interaction should definitely use the application refresh token. 

I could be wrong here.  Every time I think I understand OAuth, I find out I don't actually understand it.